librad/src/net/peer.rs

```diff
@@ -467,22 +467,18 @@ where
 /// `refs/remotes/<origin>/<path>`
 fn urn_context(local_peer_id: &PeerId, urn: Either<RadUrn, Originates<RadUrn>>) -> RadUrn {
     match urn {
-        Either::Left(urn) => urn,
+        Either::Left(urn) => RadUrn {
+            path: qualify_path(urn.path),
+            ..urn
+        },
         Either::Right(Originates { from, value }) => {
             let urn = value;
 
             if from == *local_peer_id {
                 return urn;
```

Comment:
I wrote some tests when [forward-porting](https://github.com/radicle-dev/radicle-link/commit/b302017c5d8351898b96e9f439a632ef11a489b2) this, and found that we need to qualify acc. to the same rules as for a `Left`: https://github.com/radicle-dev/radicle-link/commit/b302017c5d8351898b96e9f439a632ef11a489b2#diff-b650ce3d62e812ad5579e43d0805c510f7324e38f24762a9a81f5466b760993bR705

I also think that `qualify_path` needs to honour whether we're qualifying a remote or not. I wrote it like so: https://github.com/radicle-dev/radicle-link/commit/b302017c5d8351898b96e9f439a632ef11a489b2#diff-b650ce3d62e812ad5579e43d0805c510f7324e38f24762a9a81f5466b760993bR468

Reply:
Ach, great catch! It probably just makes sense to crib your implementation there and convert the `RefLike` to `str` and `Path::parse`. "Stand on the shoulder of giants", and all that

Reply:
Is there a reason to strip the `refs/` [here](https://github.com/radicle-dev/radicle-link/blob/b302017c5d8351898b96e9f439a632ef11a489b2/librad/src/net/peer.rs#L474), when you could start [this path](https://github.com/radicle-dev/radicle-link/blob/b302017c5d8351898b96e9f439a632ef11a489b2/librad/src/net/peer.rs#L470) with `reflike!("remotes")`?

Reply:
Ah my father's parentheses got me mixed up. You're stripping the `ref` from the qualified path.